### ISSUE TITLE
Website is very slow to load

### ISSUE THREAD

**Issue by redoper1** [Pictogrammers/pictogrammers.com#25]
I wanted to pick an icon on https://materialdesignicons.com/ and was redirected to the new website https://pictogrammers.com/library/mdi/?welcome.

But this website is unusable as it's very slow to load. I am now waiting a few minutes (approx. 8 minutes) for it to finish loading. I even got timeouts for multiple requests.

I'm located in Central Europe (Czech Republic) area and I don't know in which AWS region is this site hosted, but it appears that it is not very usable from Europe (other colleagues from different areas have same problems).

**Comment by psychonetic** [comment]
Having the same issue (Germany).

**Comment by benjaminfunk** [comment]
Same (Germany)

**Comment by droans** [comment]
Seems to be down now for me, Midwest US.

**Comment by mririgoyen** [comment]
Thanks, we're actively looking into why the site isn't responding to traffic.

**Comment by mririgoyen** [comment]
We've identified the issue and the site appears to be functioning again as expected. We'll continue to monitor it throughout the day. Please report back if you see any other issues. Thank you.

**Comment by curtchan** [comment]
Seems like the website is still slow to respond

**Comment by mririgoyen** [comment]
Yep, we're on it. It's available, but very slow everywhere. 😭

**Comment by curtchan** [comment]
Just assumption - is your PHP server dying? max requests? some request taking too long to execute/query and slows down all other responses? If so consider plugging in cloudflare, caching, cdns to ease up traffic

**Comment by mririgoyen** [comment]
It's some limitation of the tier we're on in AWS in terms of CPU usage, but we're not sure where the extra CPU usage is coming from yet. Just trying to get things up and stabilized for now.

**Comment by curtchan** [comment]
Idk, quickly checking seems like page loads twice all assets? and you dont have assets combining? (lots of requests to css/js files), just assumption, idk the tech stack you use except from webpack, which afaik allows for combining, judging by the way files are named you are not using prod build for assets (unless its just now because you are middle of debugging), good luck anyway

**Comment by ImMALWARE** [comment]
Can you please return materialdesignicons.com back? Just why did you do this? Everything has been working fine every time :(
This website never loads, "Gathering up the icons..." for 10 mins already

Do you guys maybe know any mirror or similar website?

**Comment by ImMALWARE** [comment]
Omg, it finally loaded, lol

**Comment by mririgoyen** [comment]
We've found another issue with a server module causing the CPU to spike under load, pushing us into the burst zone for our AWS host. We've disabled said module and the server has begun to stabilize, however it may be a few hours for the site to return fully to normal. We're continuing to monitor the server in real-time to be certain we've fixed the issue. Thank you for your patience.

**Comment by dayre** [comment]
Thanks for looking into it so quickly and trying to resolve it, i know how stressful it can be when a switchover like this doesn't go as planned.   Looking forward to the site working as intended.

**Comment by mririgoyen** [comment]
We continue to monitor the situation, but have not noticed an improvement. We are removing the redirection from `materialdesignicons.com` to the new site while we continue to troubleshoot.

**Comment by redoper1** [comment]
I just tried the website and the initial load time of the viewport (FullHD monitor) of the MDI icons list (with browser cache disabled) was around 3,5 seconds (tried loading 10 times). So I personally don't see any further problems at least from my point of view.

**Comment by mririgoyen** [comment]
I've identified the likely cause of the server overload... an overzealous pre-fetcher on the part of Next.js's `<Link>` component. We've merged the changes and are testing them right now. If everything looks good in the next hour or so, we will reimplement the redirect.

**Comment by mririgoyen** [comment]
We've reenabled the redirection from `materialdesignicons.com` and will be monitoring the server throughout the evening.

**Comment by mcfarljw** [comment]
It's still just spinning on "Gathering up the icons..." for me after waiting for over 5 minutes. In the meantime I've been using https://icones.js.org/collection/mdi to look things up.

**Comment by redoper1** [comment]
Yeah, I can confirm that problems with performance are partially back. Site loaded after approx. 1 minute for the first load and now when most of the assets are in the cache it's ok.

But on the first hit, it takes still a lot of time to load.

**Comment by mririgoyen** [comment]
We made a few more changes over the past couple days to help relieve extra load on the server. The last of those changes went live yesterday afternoon and the site has been stable since then. This morning, we had 150+ unique users on the site with very minimal CPU load, indicating that the underlying problem has been resolved.

Thank you all for your patience as we experienced these growing pains. I hope you all enjoy the new site now that it is usable!
